plugins/factions/plugin/sv_hooks.lua

```diff
@@ -44,6 +44,14 @@ function Factions:CharacterGenderChanged(player, char, new_gender, old_gender)
   Characters.set_model(player, player:get_faction():get_random_model(player))
 end
 
+function Factions:PreCreateCharacter(player, data)
+  local faction_table = Factions.find_by_id(data.faction)
+  if faction_table and (!data.name or !string.presence(data.name)) then
```

Comment:
Just `!string.presence(data.name)` is enough in this case. It checks if the passed argument is string internally.
Also, put a newline before this `if`.

Reply:
What if data.name equals nil?

Reply:
It will work as intended. `string.pesence(nil)` will not error, it will return nil, same as `string.presence('')`